File: Locations/MainBase/MainBase.java
package Locations.MainBase;

import Locations.Location;
import Locations.LocationHandler;
import commandAndMessage.command.CommandHandler;
import commandAndMessage.command.CommandTuple;
import commandAndMessage.command.NullCommandHandler;
import commandAndMessage.main.KeyConstants;

import java.util.Arrays;
import java.util.List;

/**
 * Created by Mark Chimes on 2017/11/08.
 */
public class MainBase extends Location {
    @Override
    public List<String> description() {
        return Arrays.asList("The main base.");
    }

    @Override
    public String name() {
        return "Main Base";
    }


    @Override
    public List<String> nextMessage() {
        return KeyConstants.EMPTY_LIST;
    }

    @Override
    public List<Location> getSublocations() {
        return Arrays.asList(new LivingQuarters(), new ControlRoom());
    }
}

File: Locations/MainBase/MainBase.kt
package Locations.MainBase

import Locations.Location
import Locations.LocationHandler
import People.Person
import commandAndMessage.command.CommandHandler
import commandAndMessage.command.CommandTuple
import commandAndMessage.command.NullCommandHandler
import commandAndMessage.main.KeyConstants

import java.util.Arrays

/**
 * Created by Mark Chimes on 2017/11/08.
 */
class MainBase(allPeople: List<Person>, sublocations: List<Location>) : Location(allPeople, sublocations) {

    override fun description(): List<String> {
        return Arrays.asList("The main base.")
    }

    override fun name(): String {
        return "Main Base"
    }


    override fun nextMessage(): List<String> {
        return KeyConstants.EMPTY_LIST
    }

}